### PR TITLE
tools/image-info: properly read systemD drop-in

### PR DESCRIPTION
**Pull request by lavocatt**
ConfigParser is not able to properly load the SystemD drop-in files. Because:

* Multiple assignments to the same option in systemd units are concatenated with white spaces.
* Multiple sections with the same section name are merged.
* An option assignment of empty value resets the option.

See https://github.com/python/cpython/issues/75709

Introduce SystemDConfigParser which has the wanted behavior.

### CODE REVIEW

**Comment by lavocatt** [comment]
Tested alongside #84 to validate that it fixes the issue with the newer OSBuild version.

**Comment by achilleas-k** [comment]
Are we going with just reading the non-comment lines for this instead?

**Comment by thozza** [comment]
> Are we going with just reading the non-comment lines for this instead?

+1 for just reading non-commented lines. TBH, this looks like an overkill to me for the purpose we use image-info.

My suggestion would be to e.g. rename https://github.com/osbuild/manifest-db/blob/e918b6adfe135c006f16cbf4976a8add0522883c/tools/image-info#L1602-L1627 to something generic and then use it in the case of unit dropins as well as for tmpfilesd configs.

**Comment by lavocatt** [comment]
> > Are we going with just reading the non-comment lines for this instead?
> 
> +1 for just reading non-commented lines. TBH, this looks like an overkill to me for the purpose we use image-info.
> 
> My suggestion would be to e.g. rename
> 
> https://github.com/osbuild/manifest-db/blob/e918b6adfe135c006f16cbf4976a8add0522883c/tools/image-info#L1602-L1627
> to something generic and then use it in the case of unit dropins as well as for tmpfilesd configs.

Why using it for `tmpfilesd` ? I feel this is out of the scope of this PR.

**Comment by thozza** [comment]
> Why using it for `tmpfilesd` ? I feel this is out of the scope of this PR.

So the function `read_tmpfilesd_config()` has literally `tmpfilesd` in its name, since it is used for reading tmpfilesd configuration files. What I'm saying is that its implementation is IMO exactly what would be sufficient also for reading systemd unit files. So in case you'll use it instead of the INI parser, I propose to rename it to something generic (like `read_uncommented_lines()`) and call it from two places (tmpfilesd config handling AND systemd unit handling).

If you want to do it fancier, you could pass characters that we used to distinguish comments as an argument and maybe use it elsewhere as well. But that is just a super-nice-to-have and out of scope IMO.

**Comment by lavocatt** [comment]
> > Why using it for `tmpfilesd` ? I feel this is out of the scope of this PR.
> 
> So the function `read_tmpfilesd_config()` has literally `tmpfilesd` in its name, since it is used for reading tmpfilesd configuration files. What I'm saying is that its implementation is IMO exactly what would be sufficient also for reading systemd unit files. So in case you'll use it instead of the INI parser, I propose to rename it to something generic (like `read_uncommented_lines()`) and call it from two places (tmpfilesd config handling AND systemd unit handling).
> 
> If you want to do it fancier, you could pass characters that we used to distinguish comments as an argument and maybe use it elsewhere as well. But that is just a super-nice-to-have and out of scope IMO.

Ok, my eyes were not in front of the eyeballs. I get you now.

File: tools/image-info [hunk 1]
```diff
@@ -1596,10 +1566,10 @@ def read_systemd_service_dropins(tree):
...
     ]
 
-    return _read_glob_paths_with_parser(tree, checked_globs, read_systemd_service_dropin)
+    return _read_glob_paths_with_parser(tree, checked_globs, read_config_file_no_comment)
 
 
-def read_tmpfilesd_config(config_path):
+def read_config_file_no_comment(config_path):
     """
     Read tmpfiles.d configuration files.
```
Comment (thozza) [inline tools/image-info]: Nitpick, it would be good to change the doc text of the function.

**Review by thozza — COMMENTED** [review]
I think that the doc text of the renamed function should be amended. The rest looks good 👍

**Review by thozza — APPROVED** [review]

**Pull request merged**